runtime/src/cost_tracker.rs

```diff
@@ -182,20 +182,26 @@ impl CostTracker {
         transaction: &SanitizedTransaction,
     ) -> Result<(), CostTrackerError> {
         // check against the total package cost
-        if self.block_cost.saturating_add(cost) > self.block_cost_limit {
-            return Err(CostTrackerError::WouldExceedBlockMaxLimit);
+        if let Some(block_cost_limit) = self.block_cost_limit {
+            if self.block_cost.saturating_add(cost) > block_cost_limit {
+                return Err(CostTrackerError::WouldExceedBlockMaxLimit);
+            }
         }
 
         // if vote transaction, check if it exceeds vote_transaction_limit
-        if transaction.is_simple_vote_transaction()
-            && self.vote_cost.saturating_add(cost) > self.vote_cost_limit
-        {
-            return Err(CostTrackerError::WouldExceedVoteMaxLimit);
+        if let Some(vote_cost_limit) = self.vote_cost_limit {
+            if transaction.is_simple_vote_transaction()
```

Comment:
I would check `is_simple_vote_transaction` before unwrapping Option. Not sure how much difference it'd make tho.